src/components/MobileWalletDialog/routes/CreateWallet/CreateBackupConfirm.tsx

```diff
@@ -0,0 +1,301 @@
+    }),
+    [borderColor],
+  )
+
+  const words = useMemo(() => {
+    if (!location.state?.vault) return []
+    return location.state.vault.getWords() ?? []
+  }, [location.state?.vault])
+
+  const randomWordIndices = useMemo(() => {
+    const indices = Array.from({ length: words.length }, (_, i) => i)
+    return indices.sort(() => Math.random() - 0.5).slice(0, 12)
+  }, [words.length])
+
+  const generateTestWords = useCallback((targetWord: string) => {
+    let randomWords = uniq(bip39.generateMnemonic(256).split(' ')) as string[]
+    const otherWords = randomWords
+      .filter(word => word !== targetWord)
+      .sort(() => Math.random() - 0.5)
+      .slice(0, 14)
+
+    const allWords = [...otherWords, targetWord] as string[]
+    return allWords.sort(() => Math.random() - 0.5)
+  }, [])
+
+  const handleWalletSelect = useCallback(
+    async (wallet: RevocableWallet) => {
+      const adapter = await getAdapter(KeyManager.Mobile)
+      const deviceId = wallet.id
+      if (adapter && deviceId) {
+        const { name, icon } = MobileConfig
+        try {
+          const wallet = await adapter.pairDevice(deviceId)
+          await wallet?.loadDevice({ mnemonic: location.state?.vault?.mnemonic ?? '' })
+
+          if (!(await wallet?.isInitialized())) {
+            await wallet?.initialize()
+          }
+          dispatch({
+            type: WalletActions.SET_WALLET,
+            payload: {
+              wallet,
+              name,
+              icon,
+              deviceId,
+              meta: { label: location.state?.vault?.label },
+              connectedType: KeyManager.Mobile,
+            },
+          })
+          dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
+          dispatch({
+            type: WalletActions.SET_CONNECTOR_TYPE,
+            payload: { modalType: KeyManager.Mobile, isMipdProvider: false },
+          })
+
+          localWallet.setLocalWallet({ type: KeyManager.Mobile, deviceId })
+          localWallet.setLocalNativeWalletName(location.state?.vault?.label ?? 'label')
+        } catch (e) {
+          console.log(e)
+        }
+      }
+    },
+    [
+      dispatch,
+      getAdapter,
+      localWallet,
+      location.state?.vault?.mnemonic,
+      location.state?.vault?.label,
+    ],
+  )
+
+  useEffect(() => {
+    if (selectedWordIndex === null && words.length > 0) {
+      setSelectedWordIndex(0)
+      const targetWord = words[randomWordIndices[0]]
+      setTestWords(generateTestWords(targetWord ?? ''))
+    }
+  }, [selectedWordIndex, words, randomWordIndices, generateTestWords])
+
+  const handleWordClick = useCallback(
+    (word: string) => {
+      const currentWordIndex = randomWordIndices[selectedWordIndex ?? 0]
+      if (words[currentWordIndex] === word) {
+        setSelectedWordIndex(prev => {
+          const next = (prev ?? -1) + 1
+          if (next >= TEST_COUNT_REQUIRED) {
+            ;(async () => {
```

Comment:
q: should this method be an async callback instead of IIAFE?

Reply:
Changed in [4dc0d6b](https://github.com/shapeshift/web/pull/8760/commits/4dc0d6b0b7dce8f4017b70bf90e62cbb5efddf1e)

Reply:
No async no problems, even better!